src/main/java/duke/utility/Ui.java

```diff
@@ -0,0 +1,231 @@
+package duke.utility;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.Scanner;
+
+import duke.exception.DukeException;
+import duke.task.DeadlineTask;
+import duke.task.EventTask;
+import duke.task.Task;
+import duke.task.ToDoTask;
+
+/**
+ * This class deals with unit interface of Duke. This includes the messages that Duke will send as well as
+ * reading the user input.
+ */
+public class Ui {
+    private static final String INDENTATION = "  ";
+    private Scanner sc;
+
+    /**
+     * Constructs a new Ui object.
+     */
+    public Ui() {
+        this.sc = new Scanner(System.in);
+    }
+
+    /**
+     * Reads the user input as a string.
+     * @return user input
+     */
+    public String readLine() {
+        return this.sc.nextLine();
+    }
+
+    /**
+     * Welcomes the user when the user starts Duke.
+     */
+    public String welcome() {
+        return "Hi! My name is Duke.\nWhat do you want me to do?";
+    }
+
+    /**
+     * Sends a farewell message to the user when the user exit Duke.
+     */
+    public String exit() {
+        return "Bye. Thank you for using me!";
+    }
+
+    /**
+     * Returns a message about the specified task has been marked as done.
+     * @param task task that was marked as done
+     * @return done success message
+     */
+    public String doneSuccess(Task task) {
+        return "Successfully marked this task as done:\n" + INDENTATION + task.toString();
+    }
+
+    /**
+     * Returns a message about the specified task has been deleted.
+     * @param task task that was deleted
+     * @param taskSize the task size after deletion
+     * @return delete success message
+     */
+    public String deleteSuccess(Task task, int taskSize) {
+        return "Okay. I will delete this task:\n" + INDENTATION + task + "\n"
+                + "Now you have " + taskSize + " " + (taskSize == 1 ? "task " : "tasks ") + "in the list.";
```

Comment:
I like how you included line breaks for long String value,  improves readability. 👍